src/main/java/jenkins/plugins/logstash/persistence/LogzioDao.java

```diff
@@ -0,0 +1,202 @@
+package jenkins.plugins.logstash.persistence;
+
+import com.github.wnameless.json.flattener.JsonFlattener;
+import com.google.gson.JsonObject;
+import io.logz.sender.FormattedLogMessage;
+import io.logz.sender.HttpsRequestConfiguration;
+import io.logz.sender.HttpsSyncSender;
+import io.logz.sender.SenderStatusReporter;
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+import io.logz.sender.exceptions.LogzioServerErrorException;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Logz.io Data Access Object.
+ *
+ * @author Ido Halevi
+ *
+ */
+
+public class LogzioDao extends AbstractLogstashIndexerDao {
+    private final String TYPE = "jenkins_logstash_plugin";
+    private String key;
+    private String host;
+    private LogzioHttpsClient httpsClient;
+
+    //primary constructor used by indexer factory
+    public LogzioDao(String host, String key){
+        this(null, host, key);
+    }
+
+    // Factored for unit testing
+    LogzioDao(LogzioHttpsClient factory, String host, String key) {
+        this.host = host;
+        this.key = key;
+        try{
+            this.httpsClient = factory == null ? new LogzioHttpsClient(key, host, TYPE) : factory;
```

Comment:
minor: move new LogzioHttpsClient(key, host, TYPE) to other constructor and make httpsClient non-nullable?

Reply:
I'm not sure I understand what do you mean, It is the same code style used in other endpoints. For example in ES: `clientBuilder = factory == null ? HttpClientBuilder.create() : factory;`.

Reply:
there is a lot of legacy code out there, if you don't want to change it though I'm not insisting